### PR TITLE
✨ Added Giggl Presence

### PR DESCRIPTION
**Pull request by OfficialCRUGG**
Hey!
This PR adds a Presence for Giggl. It dynamically shows the creator of the Portal and whether the user is also in a voice call as well as some static info for the Landing and Status Pages.

This is my first PR in ages, so I might have messed up one or another thing. If there is anything I need to change, feel free to tell me, and I'll fix it ASAP!

Images:
image
image
image
image

### CODE REVIEW

**Comment by OfficialCRUGG** [comment]
To be honest, I got no clue what exactly CodeFactor wants from me. I know what array destructuring is, but I don't know how it would be used in this context...

File: websites/G/Giggl/presence.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        }
+
+    // Presence for Giggl itself
+    } else if (document.location.hostname === "canary.giggl.app") {
+        presenceData.details = "Browsing Giggl";
+
+        // If in a portal, set status to that
+        if (document.location.pathname.startsWith("/portal")) {
+            presenceData.details = "In a Portal";
+            presenceData.state = document.querySelector("title").innerText.split(" • ")[0];
```
Comment (Bas950) [inline websites/G/Giggl/presence.ts:31]:
```suggestion
            [presenceData.state] = document.querySelector("title").innerText.split(" • ");
```
Try if that works

Comment (OfficialCRUGG) [reply websites/G/Giggl/presence.ts:31]: Thanks! Will try that out tomorrow!

**Comment by OfficialCRUGG** [comment]
Wait does this thing really want me to use 2 space Indentation?

**Comment by OfficialCRUGG** [comment]
Finally... 😂

File: websites/G/Giggl/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "description": {
+        "en": "Watch & browse the web, together - From hosting virtual movie nights, browsing the web with friends or watching anime with your long-distance partner: Giggl lets you do that together in real-time."
+    },
+    "url": [
+        "giggl.app",
+        "canary.giggl.app",
+        "status.giggl.app"
+    ],
+    "version": "1.0.0",
+    "logo": "https://i.imgur.com/pwh2bwL.jpg",
```
Comment (Bas950) [inline websites/G/Giggl/dist/metadata.json:16]: logo must be at least 512x512

**Review by Bas950 — CHANGES_REQUESTED** [review]
kinda low quality presence, is there anything else you can add to it?

**Comment by OfficialCRUGG** [comment]
Well, there would be some stuff that could be added to this in theory, but lots of stuff is not parsable from the Portal Page. For example the Portal Name or whether it's public or private. I know the Presence doesn't really do a lot, but it's still some useful information some people might want to share on their status.

As for the Logo, I can update that in a moment.

**Comment by Slowlife01** [comment]
you can still add like viewing profile

**Comment by OfficialCRUGG** [comment]
@Slowlife01 Completely forgot about that. Thanks for pointing it out. Is this good now?
image

File: websites/G/Giggl/presence.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    if (document.querySelector("svg.feather.feather-phone-missed")) {
+      presenceData.smallImageKey = "call";
+      presenceData.smallImageText = "In a call";
+    }
+
+    // Check if viewing a profile
+    if (document.querySelector("div.sc-pkHUE.fIcDpF")) {
+      presenceData.details = `Viewing ${document.querySelector("div.sc-pkHUE.fIcDpF").children[1].textContent}'s Profile`;
+      delete presenceData.state;
+    }
```
Comment (Slowlife01) [inline websites/G/Giggl/presence.ts:44]: Use `document.querySelector(".feather.feather-map-pin")` to check, and `document.querySelector("p")` for username  probably would be better

File: websites/G/Giggl/presence.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    // Check if in a call
+    if (document.querySelector("svg.feather.feather-phone-missed")) {
+      presenceData.smallImageKey = "call";
+      presenceData.smallImageText = "In a call";
+    }
+
+    // Check if viewing a profile
+    if (document.querySelector(".feather.feather-map-pin")) {
+      presenceData.details = `Viewing ${document.querySelector("div.sc-pkHUE.fIcDpF").querySelector("p").textContent}'s Profile`;
```
Comment (Slowlife01) [inline websites/G/Giggl/presence.ts:42]: I mean literally just p alone

Comment (OfficialCRUGG) [reply websites/G/Giggl/presence.ts:42]: Oh yeah. Didn't think that would just work. One sec.

**Review by Bas950 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**